<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebugWindow</class>
 <widget class="QMainWindow" name="DebugWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1135</width>
    <height>710</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1135</width>
    <height>710</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1135</width>
    <height>710</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>GBmu debugger</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>641</width>
      <height>401</height>
     </rect>
    </property>
    <property name="title">
     <string>Memory</string>
    </property>
    <widget class="QTableWidget" name="memory">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>621</width>
       <height>331</height>
      </rect>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="rowCount">
      <number>1</number>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>35</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>true</bool>
     </attribute>
     <row/>
     <column>
      <property name="text">
       <string>x0</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x1</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x2</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x3</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x4</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x5</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x6</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x7</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x8</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x9</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>xA</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>xB</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>xC</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>xD</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>xE</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>xF</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string>42</string>
      </property>
     </item>
    </widget>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>20</y>
       <width>621</width>
       <height>41</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
      <property name="spacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>11</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>480</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Inspect</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="HexSpinBox" name="inspectMemory">
        <property name="prefix">
         <string>0x</string>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>171</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>General registers</string>
    </property>
    <widget class="QTableWidget" name="generalRegisters">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>151</width>
       <height>200</height>
      </rect>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>126</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>false</bool>
     </attribute>
     <row>
      <property name="text">
       <string>PC</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>AF</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>BC</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>DE</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>HL</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>SP</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string> Value</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="1" column="0">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="2" column="0">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="3" column="0">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="4" column="0">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="5" column="0">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>120</y>
      <width>225</width>
      <height>541</height>
     </rect>
    </property>
    <property name="title">
     <string>Video registers</string>
    </property>
    <widget class="QTableWidget" name="videoRegisters">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>206</width>
       <height>500</height>
      </rect>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectItems</enum>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>80</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>false</bool>
     </attribute>
     <row>
      <property name="text">
       <string>LCDC</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>STAT</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>SCY</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>SCX</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>LY</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>LYC</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>DMA</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>BGP</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>OBP0</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>OBP1</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>WY</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>WX</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>BCPS</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>BCPD</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>OCPS</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>OCPD</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Addr</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Value</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string>FF40</string>
      </property>
      <property name="textAlignment">
       <set>AlignLeading|AlignVCenter</set>
      </property>
     </item>
     <item row="0" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="1" column="0">
      <property name="text">
       <string>FF41</string>
      </property>
     </item>
     <item row="1" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="2" column="0">
      <property name="text">
       <string>FF42</string>
      </property>
     </item>
     <item row="2" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="3" column="0">
      <property name="text">
       <string>FF43</string>
      </property>
     </item>
     <item row="3" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="4" column="0">
      <property name="text">
       <string>FF44</string>
      </property>
     </item>
     <item row="4" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="5" column="0">
      <property name="text">
       <string>FF45</string>
      </property>
     </item>
     <item row="5" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="6" column="0">
      <property name="text">
       <string>FF46</string>
      </property>
     </item>
     <item row="6" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="7" column="0">
      <property name="text">
       <string>FF47</string>
      </property>
     </item>
     <item row="7" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="8" column="0">
      <property name="text">
       <string>FF48</string>
      </property>
     </item>
     <item row="8" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="9" column="0">
      <property name="text">
       <string>FF49</string>
      </property>
     </item>
     <item row="9" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="10" column="0">
      <property name="text">
       <string>FF4A</string>
      </property>
     </item>
     <item row="10" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="11" column="0">
      <property name="text">
       <string>FF4B</string>
      </property>
     </item>
     <item row="11" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="12" column="0">
      <property name="text">
       <string>FF68</string>
      </property>
     </item>
     <item row="12" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="13" column="0">
      <property name="text">
       <string>FF69</string>
      </property>
     </item>
     <item row="13" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="14" column="0">
      <property name="text">
       <string>FF6A</string>
      </property>
     </item>
     <item row="14" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="15" column="0">
      <property name="text">
       <string>FF6B</string>
      </property>
     </item>
     <item row="15" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>90</y>
      <width>239</width>
      <height>571</height>
     </rect>
    </property>
    <property name="title">
     <string>Other registers</string>
    </property>
    <widget class="QTableWidget" name="otherRegisters">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>219</width>
       <height>530</height>
      </rect>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>80</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>false</bool>
     </attribute>
     <row>
      <property name="text">
       <string>P1</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>SB</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>SC</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>DIV</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>TIMA</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>TMA</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>TAC</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>KEY1</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>VBK</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>HDMA1</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>HDMA2</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>HDMA3</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>HDMA4</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>HDMA5</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>SVSK</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>IF</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>IE</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Addr</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Value</string>
      </property>
     </column>
     <item row="0" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="1" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="2" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="3" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="4" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="5" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="6" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="7" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="8" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="9" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="10" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="11" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="12" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="13" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="14" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="15" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
     <item row="16" column="1">
      <property name="text">
       <string>NOT IMPLEMENTED</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>10</y>
      <width>461</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Disassembler</string>
    </property>
    <widget class="QTableWidget" name="disassembler">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>441</width>
       <height>201</height>
      </rect>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderCascadingSectionResizes">
      <bool>false</bool>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>120</number>
     </attribute>
     <attribute name="horizontalHeaderHighlightSections">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>35</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string>Addr</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Opcode</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Instruction</string>
      </property>
     </column>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionStep"/>
   <addaction name="actionExecute"/>
  </widget>
  <action name="actionStep">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/png/arrow-into-drive-symbol.png</normaloff>:/icons/icons/png/arrow-into-drive-symbol.png</iconset>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionExecute">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/icons/png/arrow-in-u-shape-to-turn.png</normaloff>:/icons/icons/png/arrow-in-u-shape-to-turn.png</iconset>
   </property>
   <property name="text">
    <string>Execute</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>HexSpinBox</class>
   <extends>QSpinBox</extends>
   <header>hexspinbox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
